cabal-install/src/Distribution/Client/Main.hs

```diff
@@ -1146,13 +1144,14 @@ listAction :: ListFlags -> [String] -> Action
 listAction listFlags extraArgs globalFlags = do
   let verbosity = fromFlag (listVerbosity listFlags)
   config <- loadConfigOrSandboxConfig verbosity globalFlags
+  canonicalizedHcPath <- mapM Directory.canonicalizePath (listHcPath listFlags)
```

Comment:
Are there any other places in cabal codebase where `canonicalizePath` or anything equivalent is called? Should there be more places? Is there a way to make it more likely that all places that need it have it, e.g., defining a function that does both `listHcPath` and `canonicalizePath` and partially hiding or deprecating `listHcPath` , if that's the context where many of the `canonicalizePath` calls (should) happen.

Reply:
@Mikolaj Unfortunately I'm not familiar enough with the codebase to answer this accurately.

Reply:
yeah there's a fair amount of places where `canonicalizePath` and `canonicalizePathNoThrow` are used in cabal-install.

> Should there be more places?

no idea yet

> Is there a way to make it more likely that all places that need it have it

All CLI option data types should be built with it, yes. I feel like this is another PR however.

Reply:
> I feel like this is another PR however.

That's a good point. Maybe we could open a ticket so that we don't forget?

Reply:
Using `canonicalizePath` helped with #9578.

https://github.com/haskell/cabal/blob/50d5e31dc2e7b38a5603dea1379ae3be96608ee3/cabal-install-solver/src/Distribution/Solver/Types/ProjectConfigPath.hs#L153-L229